save-agent/src/commonMain/kotlin/com/saveourtool/save/agent/utils/HttpUtils.kt

```diff
@@ -58,13 +68,38 @@ internal suspend fun SaveAgent.processRequestToBackendWrapped(
  * @param body
  * @return result wrapping [HttpResponse]
  */
-internal suspend fun HttpClient.download(url: String, body: Any?): Result<HttpResponse> = runCatching {
-    post {
+internal suspend fun HttpClient.download(url: String, body: Any?, file: Path): Result<HttpResponse> = runCatching {
+    preparePost {
         url(url)
         contentType(ContentType.Application.Json)
         accept(ContentType.Application.OctetStream)
         body?.let { setBody(it) }
     }
+        .execute { httpResponse ->
+            if (httpResponse.status.isSuccess()) {
+                val channel: ByteReadChannel = httpResponse.body()
+                val totalBytes = AtomicLong(0L)
+                while (!channel.isClosedForRead) {
+                    val packet = channel.readRemaining(DEFAULT_HTTP_BUFFER_SIZE.toLong())
+                    while (!packet.isEmpty) {
+                        val bytes = packet.readBytes()
+                        fs.appendingSink(file, mustExist = false)
```

Comment:
just to be sure, maybe change `mustExist = false` to `mustExist = totalBytes.get() > 0`